packages/core/src/editor/BlockNoteEditor.ts

```diff
@@ -1056,6 +1056,20 @@ export class BlockNoteEditor<
     return replaceBlocks(this, blocksToRemove, blocksToInsert);
   }
 
+  /**
+   * Undo the last action.
+   */
+  public undo() {
+    this._tiptapEditor.commands.undo();
+  }
```

Comment:
This might not work with collaboration. Look into using the editor.exec API instead too